crates/turborepo-lib/src/config/turbo_config.rs

```diff
@@ -0,0 +1,581 @@
+        self.preflight.unwrap_or_default()
+    }
+
+    pub fn timeout(&self) -> u64 {
+        self.timeout.unwrap_or(DEFAULT_TIMEOUT)
+    }
+}
+
+trait ResolvedConfigurationOptions {
+    fn get_configuration_options(self) -> Result<ConfigurationOptions, ConfigError>;
+}
+
+impl ResolvedConfigurationOptions for PackageJson {
+    fn get_configuration_options(self) -> Result<ConfigurationOptions, ConfigError> {
+        match &self.legacy_turbo_config {
+            Some(legacy_turbo_config) => {
+                let synthetic_raw_turbo_json: RawTurboJSON =
+                    serde_json::from_value(legacy_turbo_config.clone())
+                        .map_err(|_| ConfigError::Anyhow(anyhow!("global_de")))?;
+                synthetic_raw_turbo_json.get_configuration_options()
+            }
+            None => Ok(ConfigurationOptions::default()),
+        }
+    }
+}
+
+impl ResolvedConfigurationOptions for RawTurboJSON {
+    fn get_configuration_options(self) -> Result<ConfigurationOptions, ConfigError> {
+        match &self.remote_cache {
+            Some(configuration_options) => {
+                configuration_options.clone().get_configuration_options()
+            }
+            None => Ok(ConfigurationOptions::default()),
+        }
+    }
+}
+
+// Used for global config and local config.
+impl ResolvedConfigurationOptions for ConfigurationOptions {
+    fn get_configuration_options(self) -> Result<ConfigurationOptions, ConfigError> {
+        Ok(self)
+    }
+}
+
+fn get_lowercased_env_vars() -> HashMap<String, String> {
+    std::env::vars()
+        .map(|(k, v)| (k.to_ascii_lowercase(), v))
+        .collect()
+}
+
+fn get_env_var_config(
+    environment: &HashMap<String, String>,
+) -> Result<ConfigurationOptions, ConfigError> {
+    let mut turbo_mapping = HashMap::new();
+    turbo_mapping.insert(String::from("turbo_api"), "api_url");
+    turbo_mapping.insert(String::from("turbo_login"), "login_url");
+    turbo_mapping.insert(String::from("turbo_team"), "team_slug");
+    turbo_mapping.insert(String::from("turbo_teamid"), "team_id");
+    turbo_mapping.insert(String::from("turbo_token"), "token");
+    turbo_mapping.insert(String::from("turbo_remote_cache_timeout"), "timeout");
+
+    // We do not enable new config sources:
+    // turbo_mapping.insert(String::from("turbo_signature"), "signature"); // new
+    // turbo_mapping.insert(String::from("turbo_preflight"), "preflight"); // new
+    // turbo_mapping.insert(String::from("turbo_remote_cache_enabled"), "enabled");
+    // // new
+
+    let mut output_map = HashMap::new();
+
+    turbo_mapping.iter().for_each(|(k, mapped)| {
+        if let Some(value) = environment.get(k) {
+            if !value.is_empty() {
+                output_map.insert(mapped.to_string(), value.to_string());
+            }
+        }
+    });
+
+    // Process signature
+    let signature = if let Some(signature) = output_map.get("signature").cloned() {
```

Comment:
I'm confused as to why we're cloning then borrowing as a str. Could we not call `as_deref` here?

Reply:
Cleaned this up.